### PR TITLE
Feature#8 Authentication with Email/Password added

### PR DESCRIPTION
**Pull request by yovanha**
# Description
What did you implemented/Why did you implemented this:
 - We've implemented a signin method with email/password.
 - We've implement a function to detect if the email is already in use.
 - Once successfully the user is created the email is uploaded to Firebase.
 
  ## Screenshots 
image
image

### CODE REVIEW

**Review by OswaldoPineda — DISMISSED** [review]
very nice code!

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: src/containers/SignUp/index.js [hunk 1]
```diff
@@ -8,20 +8,34 @@ import {
...
+        validationSchema={validationSchema}
         initialValues={{
-          name: '',
-          email: '',
-          password: '',
-          agreeTerms: false,
+          name: 'Juan',
+          email: 'juan1@example.com',
+          password: 'Juan123%',
+          agreeTerms: true,
```
[the remark this answers is not in the document]
Comment (yovanha) [reply src/containers/SignUp/index.js]: Done

File: src/constants/schemas/validationSchema.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
   email: yup.string().required(' *Correo electrónico es requerido'),
   password: yup
     .string()
-    .min(8, ({min}) => ` *Password must be at least ${min}`)
+    .min(8, ({min}) => ` *La contraseña debe tener minímo ${min} cáracteres`)
     .required(' *Contraseña es requerida')
     .matches(
       /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,})/,
-      ' *Invalid password',
+      ' *Contraseña inválida',
```
[the remark this answers is not in the document]
Comment (yovanha) [reply src/constants/schemas/validationSchema.js]: Done

File: src/constants/schemas/validationSchema.js [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
     ),
-  agreeTerms: yup.bool().oneOf([true], 'Accept Terms & Conditions is required'),
+  agreeTerms: yup
+    .bool()
+    .oneOf([true], 'Aceptar los Terminos y Condiciones es requerido'),
```
[the remark this answers is not in the document]
Comment (yovanha) [reply src/constants/schemas/validationSchema.js]: Done

File: src/constants/schemas/validationSchema.js [hunk 4]
```diff
@@ -1,15 +1,17 @@
 import * as yup from 'yup';
 
-export const signUpValidationSchema = yup.object().shape({
+export const validationSchema = yup.object().shape({
   name: yup.string().required(' *Nombre es requerido'),
   email: yup.string().required(' *Correo electrónico es requerido'),
   password: yup
     .string()
-    .min(8, ({min}) => ` *Password must be at least ${min}`)
+    .min(8, ({min}) => ` *La contraseña debe tener minímo ${min} cáracteres`)
```
[the remark this answers is not in the document]
Comment (yovanha) [reply src/constants/schemas/validationSchema.js]: Done

**Review by cortezroberto — APPROVED** [review]
LGTM

**Review by hancel-avila — APPROVED** [review]
Nice work!

**Pull request merged**